contracts/Core/StakeManager.sol

```diff
@@ -300,21 +300,22 @@ contract StakeManager is Initializable, ACL, StakeStorage, StateManager, Pause,
         uint256 _stake = stakers[stakerId].stake;
         uint256 slashPenaltyAmount = (_stake * parameters.slashPenaltyNum()) / parameters.slashPenaltyDenom();
         _stake = _stake - slashPenaltyAmount;
-        uint256 halfPenalty = slashPenaltyAmount / 2;
 
-        if (halfPenalty == 0) return 0;
+        uint256 bounty = (slashPenaltyAmount * parameters.bountyNum()) / parameters.bountyDenom();
```

Comment:
**Possible Optimisation**

Guys, 
If we see slashPenaltyNum, denom, bountyNum, denom, burnSlashNum, denom
all are being used here only, so theoretically we can combine all of them into one external call rather than 6.
I tried doing so, please check first commit, issue is with that, it reverts with ```stack to deep``` error if we do it for all 6.

now to resolve this we can combine them all in one structure, or have have only divider, rather than num or denom.
Let me know if you think we should address this.